Lib9c/Action/TransferAsset.cs

```diff
@@ -117,8 +118,86 @@ public override IAccount Execute(IActionContext context)
             }
 
             TransferAsset3.CheckCrystalSender(currency, context.BlockIndex, Sender);
+            if (state.TryGetState(Recipient, out IValue serializedStakeState))
+            {
+                bool isStakeStateOrMonsterCollectionState;
+                if (serializedStakeState is Dictionary dictionary)
```

Comment:
it might be one of cases about, why we need type tagging for states 😇

Reply:
agree.